src/pages/resultsView/enrichments/GeneBarPlot.tsx

```diff
@@ -293,8 +295,23 @@ class GenesSelection extends React.Component<IGeneSelectionProps, {}> {
         this._geneQuery = queryStr;
     }
 
+    @computed get hasUnsupportedOQL() {
+        const geneWithUnsupportedOql = _.find(this.genesToPlot, gene => {
+            if (gene.alterations && gene.alterations.length > 0) {
+                if (this.props.containerType === AlterationContainerType.COPY_NUMBER) {
+                    let geneQuery = unparseOQLQueryLine(gene).slice(0, -1)
+                    return !(geneQuery.endsWith(": AMP") || geneQuery.endsWith(": HOMEDEL"));
```

Comment:
this is backwards. we already have the OQL in parsed form, you should go through gene.alterations and see if it has any commands that are not allowed. getting into string parsing is so limited in what it can actually handle. look at oql-parser.d.ts to see what it should look like. looks like what you'd want is 
```
const allowedCnaTypes = ["AMP", "HOMDEL"];
return (alteration.alteration_type !== 'cna' || !allowedCnaTypes.includes(alteration.constr_val));
```

Reply:
yeah, even I thought about it. but not sure why I choose this way. anyways i'll give it a try and see if it works